app/controllers/api/cloud_volumes_controller.rb

```diff
@@ -59,7 +59,11 @@ def options
 
       raise BadRequestError, "No DDF specified for - #{klass}" unless klass.respond_to?(:params_for_create)
 
-      render_options(:cloud_volumes, :form_schema => klass.params_for_create(ems))
+      if params["record_id"]
```

Comment:
I'm thinking this method can look like this now:
```
def options
  if params[:ems_id]
    ems = resource_search(params[:ems_id], :ext_management_systems, ExtManagementSystem)
    raise BadRequestError, "No CloudVolume support for - #{ems.class}" unless defined?(ems.class::CloudVolume)

    klass = ems.class::CloudVolume
    raise BadRequestError, "No DDF specified for - #{klass}" unless klass.respond_to?(:params_for_create)

    render_options(:cloud_volumes, :form_schema => klass.params_for_create(ems))
  elsif params[:id]
    cloud_volume = resource_search(params[:id], :cloud_volumes, CloudVolume)
    render_options(:cloud_volumes, :form_schema => cloud_volume.params_for_update)
  else
    super
  end
end
```

Reply:
@agrare thank you for your comments. I updated all of the PRs to work like you suggested. please review the new changes